alf/algorithms/algorithm.py

```diff
@@ -280,7 +280,12 @@ def observe_for_replay(self, exp):
                     exp.state, self.train_state_spec, value_to_match=()))
 
         if self._exp_replayer is None:
-            self._set_exp_replayer(exp)
+            # Do not even create the experience replayer if the
```

Comment:
Suggest directly changing line 245-250 to return instead of raising an error.

Reply:
Sounds good. Updated.